turbo-txpool/src/tx.rs

```diff
@@ -0,0 +1,142 @@
+use crate::error::{Error, RlpResultExt};
+
+use ethereum_types::{Address, H256, U256, U64};
+
+use tiny_keccak::{Hasher, Keccak};
+
+fn keccak(input: &[u8]) -> H256 {
+    let mut keccak = Keccak::v256();
+    keccak.update(input);
+
+    let mut output = [0u8; 32];
+    keccak.finalize(&mut output);
+    H256::from(output)
+}
+
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct Tx {
+    pub hash: H256,
+    pub nonce: U256,
+    pub to: Option<Address>,
+    pub value: U256,
+    pub gas_price: U256,
+    pub gas: U256,
```

Comment:
This can also unofficially be `u64` as geth represents it as such. I'm also not opposed to renaming to `gas_limit` as it is slightly clearer.